parquet/src/main/java/org/apache/iceberg/data/parquet/BaseParquetReaders.java

```diff
@@ -397,7 +404,7 @@ public ParquetValueReader<?> primitive(
         case INT96:
           // Impala & Spark used to write timestamps as INT96 without a logical type. For backwards
           // compatibility we try to read INT96 as timestamps.
-          return new TimestampInt96Reader(desc);
+          return int96Reader(desc);
```

Comment:
Can't this call `timestampReader(desc, NANOS, ((TimestampType) primitive).isAdjustedToUTC())` instead? The primitive type is available from `desc` so that method can check and we don't need to have a factory method for INT96.

Reply:
I should assume `isAdjustedToUTC` to be true? because primitive don't have `TimestampType` and the reader was using `OffsetDateTime`, 

my understanding is if  `OffsetDateTime` is used it is with timezone and `LocalDateTime` is without timezone.

Reply:
Yes. Because the reader produces and `OffsetDateTime`, it should only be used when the expected type is compatible.